portal/src/components/Docs/DocsSideBar.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+	<div id="sidebarmain">
+		<div id="sidebar" class="docs-sidebar">
+			<!-- Version Selector -->
+			<div class="sidebar-version-selector">
+				<p class="sidebar-section">API HOST</p>
+				<span>
+					<span>{{ host.replace(/^(https?:|)\/\//, "") }}/</span>
+					<select v-model="version">
+						<option v-for="v of versions">{{ v }}</option>
+					</select>
+				</span>
+				<button class="toggle-collapse" @click="toggleNav">Expand Nav</button>
+			</div>
+			<div v-if="routeLinkNavOpen" class="sidebar-routes-big">
+				<p class="sidebar-section">ROUTES</p>
+				<div class="route-navigator">
+					<div class="route-navigator-item" v-for="route in routes" :key="route.id">
+						<a class="route-navigator-link" @click="scrollTo($event, route.id)">{{ route.summary }}</a>
+					</div>
+				</div>
+			</div>
+			<div id="routeNavigationToggle" class="sidebar-routes-small">
+				<p class="sidebar-section">ROUTES</p>
+				<div class="route-navigator">
+					<div class="route-navigator-item" v-for="route in routes" :key="route.id">
+						<a class="route-navigator-link" @click="scrollTo($event, route.id)">{{ route.summary }}</a>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+
+import { ref } from "vue";
+
+const host = import.meta.env.VITE_APP_API_REST ?? "";
+const versions = ref<string[]>(["v3"]);
+const version = ref("v3");
+
+const routeLinkNavOpen = ref(false);
+
+const toggleNav = () => {
+	routeLinkNavOpen.value = !routeLinkNavOpen.value;
+};
+
+const scrollTo = (evt: MouseEvent, id: string) => {
+	const el = document.getElementById(id);
+	if (!el) return;
+
+	el.scrollIntoView({ behavior: "smooth" });
+};
+
+defineProps<{
+	host: string;
+	versions: object;
+	routes: any;
+}>();
+
+</script>
+
+<style scoped lang="scss">
+
+@import "@style/themes.scss";
+
+#sidebarmain {
```

Comment:
Why are we using IDs on css variables, please use classes only.

Reply:
Fixed